src/sentry/incidents/endpoints/serializers.py

```diff
@@ -168,6 +169,10 @@ def validate(self, attrs):
                 )
         attrs["use_async_lookup"] = self.context.get("use_async_lookup")
         attrs["input_channel_id"] = self.context.get("input_channel_id")
+        should_validate_channel_id = self.context.get("validate_channel_id")
+        # validate_channel_id is assumed to be true unless explicitly passed as false
+        if attrs["input_channel_id"] and should_validate_channel_id is None:
```

Comment:
Probably should write this as 

```
should_validate_channel_id = self.context.get("validate_channel_id", True)
if attrs["input_channel_id"] and should_validate_channel_id:
```

In the current implementation, if you pass `False` explicitly we will still pass this check. Could be worth testing this, since CI didn't catch it.

Reply:
I like this idea, forgot about setting a default for `.get`. `test_async_lookup_outside_transaction` in `test_project_alert_rule_details` covers the path where it's being set to False in the Slack task though and that passes (it would fail due to lack of mocking the Slack endpoint if it had passed that check and attempted to validate the channel ID).

Reply:
I think we might be missing something here, since I don't see how passing `False` explicitly before would have worked, unless maybe the test passed a falsey `input_channel_id`? Anyway, not worth blocking this over